[Chen Xiaotong] iP

WuaaAj opened the pull request.

Commit:
Add Gradle support

WuaaAj committed:
Greet

WuaaAj committed:
Level-1

WuaaAj committed:
Add List

WuaaAj committed:
Mark as done

WuaaAj committed:
implement todo, event, and deadline

WuaaAj committed:
Merge remote-tracking branch 'origin/add-gradle-support'

WuaaAj committed:
modify structure

WuaaAj committed:
handle exception

WuaaAj committed:
implement delete function

WuaaAj committed:
Documentation

WuaaAj committed:
fix bugs

WuaaAj committed:
Implement save and read data

WuaaAj committed:
Teach Duke how to understand time

WuaaAj committed:
Solve conflicts

WuaaAj committed:
Revert "Solve conflicts"

This reverts commit 904f58e5122db60a0eef4d6949f3469744cf49c5.

WuaaAj committed:
Handle conflicts

WuaaAj committed:
Handle conflicts

WuaaAj committed:
handle conflicts

WuaaAj committed:
Update Duke.java

WuaaAj committed:
Revert "Update Duke.java"

This reverts commit 1944c4a169b1b29d095bba9a3b381196acec0e70.

WuaaAj committed:
Merge & fix bugs

WuaaAj committed:
Fix bugs

WuaaAj committed:
fix bugs

WuaaAj committed:
Make the code more OOP

WuaaAj committed:
Package all files into duke

WuaaAj committed:
Implement Junit

WuaaAj committed:
Implement Junit

WuaaAj committed:
Modify code styling

WuaaAj committed:
Implement find command

WuaaAj committed:
Implement find command

GnohChengYi commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+                String[] timePieces = eventPieces[1].split("at ", 2);
+                LocalDate time = LocalDate.parse(timePieces[1]);
+                Task newTask = new Event(name, time);
+                taskList.add(newTask);
+                Ui.addTask(taskList, newTask);
+            } catch (ArrayIndexOutOfBoundsException e){
+                throw new DukeException("☹ Please enter the event command in 'event [task description]/at [start time]' format");
+            }
+        } else if (command.equals("deadline")) {
+            try {
+                String[] eventPieces = pieces[1].split("/", 2);
+                String name = eventPieces[0];
+                String[] timePieces = eventPieces[1].split("by ", 2);
+                LocalDate time = LocalDate.parse(timePieces[1]);
+                Task newTask = new Deadline(name, time);
+                taskList.add(newTask);
+                taskList.add(newTask);
+            } catch (ArrayIndexOutOfBoundsException e){
+                throw new DukeException("☹ Please enter the event command in 'deadline [task description]/by [end time]' format");
+            }
+        } else if (command.equals("delete")){
+            try {
+                int index = Integer.parseInt(pieces[1]);
+                Task task = taskList.get(index - 1);
+                taskList.remove(task);
+                Ui.delete(taskList, task);
+            } catch (IndexOutOfBoundsException e){
+                throw new DukeException("☹ OOPS!!! Invalid task number");
+            }
+
+        }else {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+        }
+
+        if(command == "bye") {
+            return true;
+        } else {
+            return false;
+        }
+    }
```
> Avoid Long Methods ([source](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods))

This method is way too long. Would it be better to move the entire `if-else` statements to another function? Besides, most `else` block can be combined if you abstract out their similarities to individual command handler classes.

GnohChengYi commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+            Files.createDirectories(Paths.get("data/"));
+            File file = new File("data/duke.txt");
+
+            //Check if the file exists
```
Is this comment a bit redundant?

GnohChengYi commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+     * @param taskList a list of tasks
+     * @throws DukeException Exceptions that are possible to occur in Duke
+     */
+    public void readFile(TaskList taskList) throws DukeException {
+        try {
+            Scanner s = new Scanner(file);
+            String current, desc, name;
+            LocalDate time;
+            char type;
+            while(s.hasNext()) {
+                current = s.nextLine();
+                type = current.charAt(1);
+                if (type == 'T') {
+                    Task newTask = new Todo(current.substring(7));
+                    newTask.setCompleted(current.charAt(4)=='X');
+                    taskList.add(newTask);
+                }
+                else if (type == 'E') {
+                    desc = current.substring(7);
+                    String[] split = desc.split("->at: ", 2);
+                    name = split[0];
+                    time = LocalDate.parse(split[1]);
+                    Task newTask = new Event(name, time);
+                    newTask.setCompleted(current.charAt(4)=='X');
+                    taskList.add(newTask);
+                }
+                else {
+                    desc = current.substring(7);
+                    String[] split = desc.split("->by: ", 2);
+                    name = split[0];
+                    time = LocalDate.parse(split[1]);
+                    Task newTask = new Deadline(name, time);
+                    newTask.setCompleted(current.charAt(4)=='X');
+                    taskList.add(newTask);
+                }
+            }
+        } catch (IOException e) {
+            throw new DukeException("OOPS!!! Error occurs when reload the data");
+        }
+    }
```
This method is too long. Would it be better to move each `if` / `else-if` block to its respective helper methods?

GnohChengYi commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,32 @@
+    private ArrayList<Task> taskList;
+
+    public TaskList() {
+        this.taskList = new ArrayList<Task>();
```
I like how you state the type (`Task`) explicitly, instead of shorten it to `new ArrayList<>()`.

GnohChengYi reviewed (commented):
Overall, I found your code easy to read for the most part except a few places where the methods/functions are too long. Consider splitting into smaller classes/methods/functions if that makes sense. There are many minor coding standard violations too, especially in the Javadocs.

tsinyee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+ * Deals with making sense of the user command
+ */
+public class Parser {
+    /**
+     * Deals with parsing the command and do some reaction
+     * @param input user command input
+     * @param taskList a list of tasks
+     * @return if the command required to exit
+     * @throws DukeException exceptions that are possible to occur in Duke
+     */
```
For Javadoc, there should be an empty line between description and parameter section

tsinyee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+                Ui.done(task);
+            } catch (IndexOutOfBoundsException e) {
+                throw new DukeException("☹ OOPS!!! Invalid task number");
+            }
+
```
consider removing this new line to be more consistent

tsinyee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+                throw new DukeException("☹ OOPS!!! Invalid task number");
+            }
+
+        }else {
```
There should be a white space before `else`

tsinyee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+                Task task = taskList.get(index - 1);
+                taskList.remove(task);
+                Ui.delete(taskList, task);
+            } catch (IndexOutOfBoundsException e){
```
There should be a white space after `(IndexOutOfBoundsException e)`

tsinyee commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,86 @@
+            } catch (ArrayIndexOutOfBoundsException e){
+                throw new DukeException("☹ Please enter the event command in 'deadline [task description]/by [end time]' format");
+            }
+        } else if (command.equals("delete")){
```
There should be a white space after `(command.equals("delete"))`

tsinyee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+     * @return a new 'Storage' object
+     * @throws DukeException Exception of Duke system
+     */
+
```
Redundant blank line between documentation block and method

tsinyee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,104 @@
+import java.nio.file.Paths;
+import java.time.LocalDate;
+import java.util.Scanner;
+
```
Missing descriptive header comment for public class

tsinyee commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,39 @@
+package duke;
+
+/**
+ * General category
```
Can be more specific for class header comments

tsinyee commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * Mark the task completed
+     */
+    public void doneTask() {
```
Setter methods for boolean variables must be of the form:
`void setFound(boolean isFound);`
```suggestion
    public void markTaskAsDone() {
```

tsinyee commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class Task {
+    private String name;
+    private Boolean completed = false;
```
Boolean variables/methods should be named to sound like booleans.
```suggestion
    private Boolean isCompleted = false;
```

tsinyee commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    /**
+     * Add task to taskList
```
In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.)

tsinyee reviewed (commented):
Some missing classes and method comments and also missing white spaces between code. 
Can consider grouping classes into packages.

WuaaAj committed:
Update README.md

WuaaAj committed:
Build Gradle

WuaaAj committed:
Merge branch-A-Gradle

WuaaAj committed:
Delete emoji that fail to compile

WuaaAj committed:
Implement GUI

WuaaAj committed:
Modify to meet code standard

WuaaAj committed:
Add assertions

WuaaAj committed:
Merge pull request #1 from WuaaAj/branch-A-Assertions

Merge branch-A-Assertions

WuaaAj committed:
Improve code quality

WuaaAj committed:
Merge pull request #2 from WuaaAj/branch-A-CodeQuality

Improve code quality

WuaaAj committed:
Implement update function

WuaaAj committed:
Polish GUI

WuaaAj committed:
Add screenshot of gui

WuaaAj committed:
Update user guide

WuaaAj committed:
Update README.md

WuaaAj committed:
Improve code quality

There are some parts do not meet
Code Standard.

Some dead code in `testDeadline`
The tense of some header comments not correct.

Delete all dead code and modify the header comments to meet code standard.

Just simply modify the code according to the issues suggested in the github

WuaaAj committed:
Improve code quality

There are some parts do not meet
Code Standard.

The try-catch block didn't follow the egyptian style braces.

Uses eqyptian style braces instead.

Makes `catch` and the end braces in the same line.

WuaaAj committed:
Update README.md
